### PR TITLE
Using docker compose to run acceptance tests

### PR DESCRIPTION
**Pull request by ollyjshaw**

### CODE REVIEW

File: _cucumber/features/support/env.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@
 Capybara.configure do |config|
 
   if ENV['HOST_TO_TEST'] == ''
-    host_to_test = "http://docker:#{ENV['AWESTRUCT_HOST_PORT']}"
+    host_to_test = "http://developers.redhat.com"
```
Comment (paulrobinson) [inline _cucumber/features/support/env.rb]: Did you mean to hard-code this URL in here?

**Comment by paulrobinson** [comment]
@LightGuard can you review please? In particular, take a look at the changes to '_docker/drupal/Dockerfile'. Olly was having problems building the Drupal image.

File: Rakefile [hunk 2]
```diff
@@ -408,10 +415,17 @@ def run_awestruct(args)
 
   if ENV['site_base_path']
     base_url = ENV['site_base_path']
-    base_url = "#{base_url}/#{ENV['site_path_suffix']}" if ENV['site_path_suffix']
+
+    if ENV['site_path_suffix'].to_s != ''
```
Comment (LightGuard) [inline Rakefile]: Could change this to a `chop.empty?`

File: Rakefile [hunk 3]
```diff
@@ -408,10 +415,17 @@ def run_awestruct(args)
...
+      base_url = "#{base_url}/#{ENV['site_path_suffix']}"
+    end
   end
+
   args ||= "" # Make sure that args is initialized
-  args << " --url " + base_url if base_url
+
+  if base_url.to_s != ''
```
Comment (LightGuard) [inline Rakefile]: Same here

**Comment by LightGuard** [comment]
It'll be fine, when I get my drupal stuff pushed I'll fix the drupal docker container.

File: Rakefile (hunk 2 above)
Comment (ollyjshaw) [inline Rakefile]: The idea here is to protect against nils and empty strings. 

```ruby
ENV['NOT_THERE'].chop
```

Will give an ```NoMethodError: undefined method `chop' for nil:NilClass``` if that environment variable is nil.

To make sure it does't blow up we convert to string as, 
```ruby
nil.to_s
```
returns ```""```

I do think using ```empty?``` makes the intent more clear, so I'll make that change now thanks.

File: _cucumber/features/support/env.rb (hunk 1 above)
Comment (ollyjshaw) [inline _cucumber/features/support/env.rb]: Yes, if running through the control script then a host is normally set. If not we could baulk, or use the live site. What do you think?

**Comment by ollyjshaw** [comment]
@paulrobinson let's try again

**Pull request merged**